pyquil/pragmas.py

```diff
@@ -0,0 +1,53 @@
+##############################################################################
+# Copyright 2016-2018 Rigetti Computing
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+##############################################################################
+
+from typing import List, Iterable
+
+from pyquil.quilbase import AbstractInstruction, Pragma
+from pyquil.quil import Program
+
+
+class InitialRewiring(Pragma):
+    def __init__(self, rewiring: str):
+        self.rewiring = rewiring
+
+    def out(self) -> str:
+        return f'PRAGMA INITIAL_REWIRING "{self.rewiring}"'
+
+
+NAIVE_REWIRING = InitialRewiring("NAIVE")
+RANDOM_REWIRING = InitialRewiring("RANDOM")
+GREEDY_REWIRING = InitialRewiring("GREEDY")
+PARTIAL_REWIRING = InitialRewiring("PARTIAL")
+
+
+class PreserveBlock(Pragma):
```

Comment:
Could this get tricky? Since it wraps the instructions, it might introduce cases where code that assumed `program.instructions` would be the whole program would break when it's only a single-element list, and in each place it will have to account for this.

Might be simpler to just vend the pragmas for block start and end separately?